lib/src/room.dart

```diff
@@ -1401,16 +1413,22 @@ class Room {
       ''');
     }
 
-    final matrixEvents = await client.getMembersByRoom(id);
+    final matrixEvents = await client.getMembersByRoom(
+      id,
+      at: at,
+      membership: membership,
+      notMembership: notMembership,
+    );
     final users = matrixEvents
             ?.map((e) => Event.fromMatrixEvent(e, this).asUser)
             .toList() ??
         [];
 
     if (cache) {
       for (final user in users) {
-        setState(user); // at *least* cache this in-memory
+        setState(user);
       }
+      await client.database?.storeUsers(users, this);
```

Comment:
we should try/catch here to prevent function break when caching failed.

Reply:
updated